app/models/MessageModel.php

```diff
@@ -50,6 +55,37 @@ function exists($id) {
       return ($this->get($id) !== NULL);
     }
 
+    function getNumUnread($studentOrRecruiterId) {
+      if (StudentModel::exists($studentOrRecruiterId)) {
+        return StudentModel::getNumUnread(new MongoId($studentOrRecruiterId));
+      } else if (RecruiterModel::IDexists($studentOrRecruiterId)) {
+        return RecruiterModel::getNumUnread(new MongoId($studentOrRecruiterId));
```

Comment:
As for the separate functions here, we could have both RecruiterModel and StudentModel extend a UserModel that shares this function for better OO design.